### PR TITLE
feat: show address reuse warning

### LINKED ISSUE #404: Show address reuse warning
Currently, it is not only possible to reuse addresses in Jam, there is also no indication that there's anything wrong with that. We probably want to, at the very least, show a warning. We should be able to easily do this for all "internal" Jam addresses, i.e. if you accidentally paste the same address twice (e.g. to fund a jar with multiple transactions).

It would also be good to show a warning in general, for any address, but we can't query the backend for that as of now.

We could also think of a "strict" setting that doesn't allow any reuse at all, no matter what.

### PR DESCRIPTION
**Pull request by dnlggr**
Resolves #404.

Warns the user in case a destination address on the Send screen is a known "internal" address that's not labelled as `new`.

Edit: Currently, we'd not allow sending to that address. Alternatively, we could also just show the warning but allow sending Any preferences @dergigi @theborakompanioni?

## 📸 

Screenshot 2022-07-20 at 11 17 05

### CODE REVIEW

File: src/components/Send.jsx [hunk 1]
```diff
@@ -266,6 +266,7 @@ export default function Send() {
   const [isSweep, setIsSweep] = useState(false)
   const [destinationJarPickerShown, setDestinationJarPickerShown] = useState(false)
   const [destinationJar, setDestinationJar] = useState(null)
+  const [destinationIsReusedAddress, setDesitnationIsReusedAddress] = useState(false)
```
Comment (theborakompanioni) [inline src/components/Send.jsx:269]: Small typo: `setDesitnationIsReusedAddress` -> `setDestinationIsReusedAddress`

File: src/components/Send.jsx [hunk 2]
```diff
@@ -425,6 +426,17 @@ export default function Send() {
     return () => abortCtrl.abort()
   }, [isOperationDisabled, wallet, reloadCurrentWalletInfo, reloadServiceInfo, loadConfigValue, t])
 
+  useEffect(() => {
+    if (walletInfo?.addressSummary[destination]) {
```
Comment (theborakompanioni) [inline src/components/Send.jsx:430]:
```suggestion
    if (destination && walletInfo?.addressSummary[destination]) {
```
?

Apparently..
```
var dict = { null: 1 }
dict[null]
// yields 1
```

Comment (dnlggr) [reply src/components/Send.jsx:430]: Yeah but will `addressSummary` ever have a `null` entry we'd accidentally access that way?

Comment (theborakompanioni) [reply src/components/Send.jsx:430]: I guess not, just trying to come up with stuff that breaks things. Maybe in the future "null" is a valid Bitcoin invoice? : D
Chances are for that are basically zero, yes. I'd say it is just good practice (apparently, `{"null": 1}[null]` yields also `1`).
I am with you, though, it's a nit.

Comment (dnlggr) [reply src/components/Send.jsx]: Added the null check, you're right it's good practice.

> Maybe in the future "null" is a valid Bitcoin invoice? : D

If that'd be the case, then we'd want to check if the `null` address is reused though right? :D

Comment (theborakompanioni) [reply src/components/Send.jsx]: > If that'd be the case, then we'd want to check if the `null` address is reused though right? :D
`null` is falsy, `"null"` is truthy, oh my..  : D

**Review by theborakompanioni — APPROVED** [review]
Approved. :muscle: 

I guess we can do the same on the Jam page in another PR.

**Comment by dergigi** [comment]
> Currently, we'd still allow sending to that address. Alternatively, we could also disable sending in this case. Any preferences @dergigi @theborakompanioni?

+1 for disabling.

**Pull request merged**